Implement javascript handlers for UI interactions

patrickjfl opened the pull request:
## Changes in this PR

Implements JS handlers for the location selection menu, pollutant selection menu, and toggling guidance for predictions. Trello ticket: https://trello.com/c/1IWw81ee/83-implement-js-handlers-for-ui-interactions

## Screenshots of UI changes

![Screenshot 2024-10-18 at 15 06 32](https://github.com/user-attachments/assets/97d786f0-1dcb-4bd9-8fd1-ebc1c006bb66)

![Screenshot 2024-10-18 at 15 07 52](https://github.com/user-attachments/assets/b429b8aa-8a04-47a3-b40b-4598206ed500)

### Before

### After

## Next steps

- [ ] Run any necessary
      [accessibility testing](https://playbook.dxw.com/guides/web-accessibility.html)
      on this feature.

patrickjfl committed:
Add JS handler for location selector

Also removes the blue ticks from the dropdown, which are probably redundant for our single-select use case.

patrickjfl committed:
Add JS handler for pollutant selector

patrickjfl committed:
Add chevron up icon

patrickjfl committed:
Update prediction component to toggle hide/display guidance and down/up chevrons

patrickjfl commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Implement JS handlers for UI interactions](https://trello.com/c/1IWw81ee/83-implement-js-handlers-for-ui-interactions)

edavey commented:
Nicely done. Just a few small questions (see screenshots):

1. should we go a bit further with the "selection-made" event handler? to close the drop down and display the selected option? 
2. are we planning to retain the blue "checks" or "ticks" in the zone selector? Perhaps they are most useful for a multi-select list, which is not the case here? So perhaps we could just lose the ticks?
3. the control for expanding / collapsing the prediction details ought to be right aligned I think?

<img width="567" alt="close_on_select" src="https://github.com/user-attachments/assets/9c0fa7c0-c4ef-4a5e-a23e-1d9948814440">

<img width="585" alt="tick_against_selection" src="https://github.com/user-attachments/assets/9fedb2b3-b0fc-4a21-95d2-c3e226b12165">

![chevrons-right-aligned](https://github.com/user-attachments/assets/59f124b9-6cb0-4033-94c5-0c9b03034f13)

patrickjfl committed:
Update Tailwind config file to include classes used in components directory

patrickjfl commented:
> Nicely done. Just a few small questions (see screenshots):
> 
>     1. should we go a bit further with the "selection-made" event handler? to close the drop down and display the selected option?
> 
>     2. are we planning to retain the blue "checks" or "ticks" in the zone selector? Perhaps they are most useful for a multi-select list, which is not the case here? So perhaps we could just lose the ticks?
> 
>     3. the control for expanding / collapsing the prediction details ought to be right aligned I think?
> 
> 
> <img alt="close_on_select" width="567" src="https://private-user-images.githubusercontent.com/20245/378349301-9c0fa7c0-c4ef-4a5e-a23e-1d9948814440.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MDc4NzcsIm5iZiI6MTcyOTUwNzU3NywicGF0aCI6Ii8yMDI0NS8zNzgzNDkzMDEtOWMwZmE3YzAtYzRlZi00YTVlLWEyM2UtMWQ5OTQ4ODE0NDQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDEwNDYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ5ZTc3NGMwMTRmZGU5NmE2M2U4MjkyZGIyZmYxZmZjZmFhZDdhNzg5NmQ5NGUwNjQxMDY1OGZhNWU3OGUwNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eNOvfrkZMFSxBRQ9fQw10sXCHKc5w-0sMWZJ-68GyZY"> <img alt="tick_against_selection" width="585" src="https://private-user-images.githubusercontent.com/20245/378349367-9fedb2b3-b0fc-4a21-95d2-c3e226b12165.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MDc4NzcsIm5iZiI6MTcyOTUwNzU3NywicGF0aCI6Ii8yMDI0NS8zNzgzNDkzNjctOWZlZGIyYjMtYjBmYy00YTIxLTk1ZDItYzNlMjI2YjEyMTY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDEwNDYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3ZTdjZGRmMDg4ZTA1MGI1ZmMyMGI1YmQxMjhhZWRkMDRmNTc4MWRhNzA0NTgyZjYyNjBjNjllNzM5YTRlOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Om2_lJmu6pYs7ZVGzxfzwdYsZ3aZ97JaMhW_2laPVBE">
> 
> ![chevrons-right-aligned](https://private-user-images.githubusercontent.com/20245/378349410-59f124b9-6cb0-4033-94c5-0c9b03034f13.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MDc4NzcsIm5iZiI6MTcyOTUwNzU3NywicGF0aCI6Ii8yMDI0NS8zNzgzNDk0MTAtNTlmMTI0YjktNmNiMC00MDMzLTk0YzUtMGM5YjAzMDM0ZjEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDEwNDYxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkYjY0ZmJiZmE3MzMzYzQ1OGI2NWM2NWIxM2QwZWEzYWNiZmEwNjRlNmFiNjU4MzQ0MmMyZGU4YTJjMzlmM2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1fizHBaVI6Pe8vZ7SY8bbw1YYLaKXyrWKfmR7xe4HVU)

good points, all actioned i believe :)

edavey committed:
Rename uv IDs for JS controls

Once we've got a comprehensive integration test for the new
styled forecasts page (`/styled_forecast`), including the air
quality alerts, we will refactor the prediction
classes

- rename `#name` to be a single word ("UV")

- move the display name ("Ultraviolet rays (UV)") to translation
file fron where it can be accessed by `PredictionComponent`

- remove `#guidance`. This belongs in the new
`PredictionComponent`
